File: Call-Android/app/src/main/java/org/ar/call/SettingActivity.java
package org.ar.call;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;

import com.gyf.immersionbar.ImmersionBar;
import com.kongzue.dialog.interfaces.OnMenuItemClickListener;
import com.kongzue.dialog.v3.BottomMenu;

import org.ar.call.p2p.VideoActivity;
import org.ar.call.multi.MultiCallActivity;
import org.ar.call.utils.SpUtil;
import org.ar.rtm.RemoteInvitation;
import org.json.JSONException;
import org.json.JSONObject;

public class SettingActivity extends BaseActivity  {

    private TextView tvXY,tvFrame;
    private Button mBtnCamera,mBtnMicrophone;
    private boolean isOpenCamera,isOpenMicrophone;
    private boolean isP2P;
    private LinearLayout llSettingResolution,llSettingCamera;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_setting);
        Intent intent =getIntent();
        isP2P =intent.getBooleanExtra("isP2P",false);
        ImmersionBar.with(this).statusBarDarkFont(true,0.2f).statusBarColor(R.color.white).init();
        init();
    }

    private void init(){
        tvXY = findViewById(R.id.tv_xy);
        tvFrame = findViewById(R.id.tv_frame);
        mBtnCamera =findViewById(R.id.setting_camera);
        mBtnMicrophone =findViewById(R.id.setting_microphone);
        llSettingResolution =findViewById(R.id.ll_setting_resolution);
        llSettingCamera =findViewById(R.id.ll_setting_camera);
        if (isP2P){
            llSettingResolution.setVisibility(View.VISIBLE);
            llSettingCamera.setVisibility(View.GONE);
            refresh();
        }else {
            llSettingResolution.setVisibility(View.GONE);
            llSettingCamera.setVisibility(View.VISIBLE);
            isOpenCamera =SpUtil.getBoolean("isOpenCamera",true);
            isOpenMicrophone =SpUtil.getBoolean("isOpenMicrophone",true);
            if (isOpenCamera){
                mBtnCamera.setBackgroundResource(R.drawable.open);
            }else {
                mBtnCamera.setBackgroundResource(R.drawable.close);
            }
            if (isOpenMicrophone){
                mBtnMicrophone.setBackgroundResource(R.drawable.open);
            }else {
                mBtnMicrophone.setBackgroundResource(R.drawable.close);
            }
        }

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }

    public void openCamera(View view){
        if (isOpenCamera){
            mBtnCamera.setBackgroundResource(R.drawable.close);
        }else {
            mBtnCamera.setBackgroundResource(R.drawable.open);
        }
        isOpenCamera =!isOpenCamera;
        SpUtil.putBoolean("isOpenCamera",isOpenCamera);
    }

    public void closeMicrophone(View view){
        if (isOpenMicrophone){
            mBtnMicrophone.setBackgroundResource(R.drawable.close);
        }else {
            mBtnMicrophone.setBackgroundResource(R.drawable.open);
        }
        isOpenMicrophone =!isOpenMicrophone;
        SpUtil.putBoolean("isOpenMicrophone",isOpenMicrophone);
    }

    private void refresh() {
        int spFrame = SpUtil.getInt("frame");
        if (spFrame == 1){
            tvFrame.setText("15");
        }else if (spFrame ==2){
            tvFrame.setText("24");
        }else if (spFrame == 3){
            tvFrame.setText("30");
        }
        int spSize = SpUtil.getInt("size");
        if (spSize == 1){
            tvXY.setText("240 * 320");
        }else if (spSize ==2){
            tvXY.setText("480 * 640");
        }else if (spSize == 3){
            tvXY.setText("720 * 1280");
        }
    }

    public void Back(View view) {
        finish();
    }

    public void FenBianLv(View view) {
        BottomMenu.show(this, new String[]{"240 * 320", "480 * 640", "720 * 1280"}, new OnMenuItemClickListener() {
            @Override
            public void onClick(String text, int index) {
                SpUtil.putInt("size",index+1);
                refresh();
                Toast.makeText(SettingActivity.this,"设置成功",Toast.LENGTH_SHORT).show();
            }
        }).setTitle("请选择分辨率");
    }

    public void Frame(View view) {
        BottomMenu.show(this, new String[]{"15", "24", "30"}, new OnMenuItemClickListener() {
            @Override
            public void onClick(String text, int index) {
                SpUtil.putInt("frame",index+1);
                refresh();
                Toast.makeText(SettingActivity.this,"设置成功",Toast.LENGTH_SHORT).show();
            }
        }).setTitle("请选择帧率");

    }


    @Override
    public void onRemoteInvitationReceived(RemoteInvitation remoteInvitation) {
            runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    try {
                        JSONObject jsonObject = new JSONObject(remoteInvitation.getContent());
                        boolean isConference = jsonObject.getBoolean("Conference");
                        if (!isConference){
                            Intent i = new Intent(SettingActivity.this, VideoActivity.class);
                            i.putExtra("RecCall",true);
                            startActivity(i);
                            finish();
                        }else {
                            Intent i = new Intent(SettingActivity.this, MultiCallActivity.class);
                            i.putExtra("RecCall",true);
                            startActivity(i);
                            finish();
                        }
                    } catch (JSONException e) {
                        e.printStackTrace();
                    }
                }
            });
    }

}

File: Call-Android/app/src/main/java/org/ar/call/SettingActivity.kt
package org.ar.call

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import com.gyf.immersionbar.ImmersionBar
import com.kongzue.dialog.v3.BottomMenu
import org.ar.call.databinding.ActivityMainBinding
import org.ar.call.databinding.ActivitySettingBinding
import org.ar.call.multi.MultiCallActivity
import org.ar.call.p2p.VideoActivity
import org.ar.call.utils.Constans
import org.ar.call.utils.SpUtil
import org.ar.rtm.RemoteInvitation
import org.json.JSONException
import org.json.JSONObject

class SettingActivity : BaseActivity() {
    private lateinit var binding:ActivitySettingBinding
    private var isSingleMode = false //是否为p2p呼叫进来的

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySettingBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)
        ImmersionBar.with(this).statusBarDarkFont(true, 0.2f).statusBarColor(R.color.white).init()
        val intent = intent
        isSingleMode = intent.getBooleanExtra(Constans.KEY_SINGLE_CALL, false)
        init()
        binding.swCamera.setOnCheckedChangeListener { _, isChecked ->
            SpUtil.putBoolean(Constans.KEY_OPEN_CAMERA, isChecked)
        }
        binding.swMic.setOnCheckedChangeListener { _, isChecked ->
            SpUtil.putBoolean(Constans.KEY_OPEN_MIC, isChecked)
        }
        binding.swDenoise.setOnCheckedChangeListener { _, isChecked ->
            SpUtil.putBoolean(Constans.OPEN_DENOISE, isChecked)
            RtcManager.instance.enableAIDeNoise(if(isChecked) 1 else 0)
        }
    }

    private fun init() {
        if (isSingleMode){
            binding.llVideoConfig.visibility = View.VISIBLE
            binding.llVideoSetting.visibility = View.GONE
            setConfig()
        }else{
            binding.llVideoConfig.visibility = View.GONE
            binding.llVideoSetting.visibility = View.VISIBLE
            setAVStatus()
        }
        binding.swDenoise.isChecked = SpUtil.getBoolean(Constans.OPEN_DENOISE)
    }





    private fun setConfig() {
        when (SpUtil.getInt(Constans.KEY_FRAME)) {
            1 -> {
                binding.tvFrame.text = "15"
            }
            2 -> {
                binding.tvFrame.text = "24"
            }
            3 -> {
                binding.tvFrame.text = "30"
            }
        }
        when (SpUtil.getInt(Constans.KEY_DIMENS)) {
            1 -> {
                binding.tvXy.text = "240 * 320"
            }
            2 -> {
                binding.tvXy.text = "480 * 640"
            }
            3 -> {
                binding.tvXy.text = "720 * 1280"
            }
        }
    }

    private fun setAVStatus(){
        binding.swMic.isChecked = SpUtil.getBoolean(Constans.KEY_OPEN_MIC, true)
        binding.swCamera.isChecked = SpUtil.getBoolean(Constans.KEY_OPEN_CAMERA, true)
    }

    override fun onRemoteInvitationReceived(remoteInvitation: RemoteInvitation) {
        runOnUiThread {
            val jsonObject = JSONObject(remoteInvitation.content)
            val isConference = jsonObject.getBoolean("Conference")
            Intent().apply {
                putExtra("RecCall",true)
                setClass(this@SettingActivity,if (isConference){
                    MultiCallActivity::class.java
                }else{
                    VideoActivity::class.java
                })
                startActivity(this)
            }
        }
    }

    fun back(view: View?) {
        finish()
    }

    public fun frameSetting(view: View?) {
        BottomMenu.show(this, arrayOf("15", "24", "30")) { text, index ->
            SpUtil.putInt(Constans.KEY_FRAME, index + 1)
            setConfig()
            Toast.makeText(this@SettingActivity, "设置成功", Toast.LENGTH_SHORT).show()
        }.title = "请选择帧率"
    }

    public fun videoDimensions(view: View?) {
        BottomMenu.show(this, arrayOf("240 * 320", "480 * 640", "720 * 1280")) { text, index ->
            SpUtil.putInt(Constans.KEY_DIMENS, index + 1)
            setConfig()
            Toast.makeText(this@SettingActivity, "设置成功", Toast.LENGTH_SHORT).show()
        }.title = "请选择分辨率"
    }
}